### DISCUSSION TITLE
How to force software rendering?

### DISCUSSION THREAD

**Discussion by NoelJacob** [marc2332/freya#536] [Q&A] [answered]
Same as the title :)

**Comment by marc2332** [comment]
Afaik Skia (the library I use in Freya) does software rendering already, why do you ask, you got some issue?

**Reply by NoelJacob** [reply]
No issue to add. I saw in README that it supports Skia and Dioxius. I couldn't find any kind of config for selecting if GPU or only software is used. I wanted to know if it works via software rendering exclusively.

**Reply by marc2332** [reply]
Ah! Well, Skia is mostly software rendering although it has some GPU acceleration features, which I have yet to see how to integrate :smile:

**Reply by marc2332** [reply]
Skia is also the only renderer Freya supports at the moment (with no plan to change that)

**Reply by NoelJacob** [reply]
So I'm assuming it's exclusively CPU for now? I'm asking bc I don't know. Also why didn't you use TinySkia. Is it bc it's about 30-100% slower than Skia?

**Reply by NoelJacob** [reply]
I'm asking because I want to run on really low end devices with Ram constrains and without a GPU.

**Reply by marc2332** [reply]
It isn't just about being slower, it has way less features:
Missing features:
```
GPU rendering.
Text rendering (maybe someday).
PDF generation.
Non-RGBA8888 images.
Non-PNG image formats.
Advanced Bézier path operations.
Conic path segments.
Path effects (except dashing).
Any kind of resource caching.
ICC profiles.
```

**Reply by marc2332** [reply]
Mmm I see, I haven't tried running skia in a very very low end device myself so I can't really give any guidance but maybe you can try and see how it goes :eyes:

**Reply by NoelJacob** [reply]
> So I'm assuming it's exclusively CPU for now?

**Reply by marc2332** [reply, marked as answer]
Yes, I think so

**Reply by marc2332** [reply]
I am pretty sure I was wrong, Skia (as configured by Freya) does indeed use GPU.
